lib/core/hashtheplanet.py

```diff
@@ -65,37 +80,16 @@ def compute_hashs(self):
         Computes all hashs.
         """
         try:
-            with open(self._input_file, "r", encoding="utf-8") as file_descriptor:
+            with open(self._input_file, "r", encoding="utf-8", newline="") as file_descriptor:
                 logger.info(f"Start reading {self._input_file}")
                 csv_reader = reader(file_descriptor)
-                header = next(csv_reader)
-                csv_reader = reader(file_descriptor)
-
-                if header:
-                    for row in csv_reader:
-                        with tempfile.TemporaryDirectory() as tmp_dir_name:
-                            technology, url = row
-                            repository = self._git_resource.clone_repository(technology, url, tmp_dir_name)
-
-                            logger.debug("Retrieving tags ...")
-                            path = f"{tmp_dir_name}/{technology}"
-                            git_tags = self._git_resource.get_tags(repository)
-                            logger.debug(f"Git tags : {git_tags}")
-                            logger.debug(f"Inserting tags for {technology} ...")
-
-                            with self._session.begin() as session: # pylint: disable=no-member
-                                self._database.insert_tags(session, technology, git_tags)
-
-                            logger.debug(f"Retrieving tags from database for {technology}")
-
-                            with self._session.begin() as session: # pylint: disable=no-member
-                                tags = self._database.get_tags(session, technology)
-                                logger.debug(f"Database tags : {tags}")
 
-                                for tag in tags:
-                                    logger.debug(f"Checkout and compute hashs for tag {tag} ...")
-                                    self._git_resource.checkout_and_compute(
-                                        session, path, repository, tag.tag)
+                for row in csv_reader:
+                    if not row:
+                        logger.warning("Input file is empty or contains an empty line")
+                    else:
+                        url = row[0]
+                        self._git_resource.clone_checkout_and_compute_hashs(self.session_scope, url)
```

Comment:
Is it possible to do this check before the for loop ?

Reply:
I changed the warning message to be clearer